contracts/interchain-token-service/src/contract/execute.rs

```diff
@@ -217,169 +255,158 @@ pub fn set_chain_config(
     }
 }
 
-/// Applies various checks on the message.
-fn apply_checks(
+fn apply_transfer(
```

Comment:
I generally don't like comments that reiterate what the code is doing, because it's too easy to drift apart. If it's not clear what's going on the code should be improved. Happy to here other opinions